### PR TITLE
[msbuild] Unify handling of Sdks.

### PR DESCRIPTION
**Pull request by rolfbjarne**
Create an Sdks class in Xamarin.MacDev.Tasks.Core, which handles both Xamarin.iOS
and Xamarin.Mac. Refactor the MacOSXSdks and IPhoneSdks classes to use the new Sdks
class.

This makes it possible to avoid some code duplication in MacOSXSdks and IPhoneSdks,
and also share code elsewhere.

This requires a bump of Xamarin.MacDev.

New commits in xamarin/Xamarin.MacDev:

* xamarin/Xamarin.MacDev@fae0237 [Xamarin.MacDev] Add GetAppleDTSettings and GetSdkSettings to the IAppleSdk interface. (#85)

Diff: https://github.com/xamarin/Xamarin.MacDev/compare/f665e3a0fccce24b9dbb31650f99ef2c73ba4d46..fae0237704b792004adf592d8c9038a9d46aafff

### CODE REVIEW

File: msbuild/Xamarin.MacDev.Tasks.Core/Sdks.cs [hunk 1]
```diff
@@ -65,5 +67,30 @@ public static AppleSdk GetSdk (string targetFrameworkMoniker)
...
+			case ApplePlatform.WatchOS:
+				return Watch;
+			case ApplePlatform.TVOS:
+				return TVOS;
+			case ApplePlatform.MacOSX:
+				return MacOS;
+			case ApplePlatform.MacCatalyst:
+				return IOS;
+			default:
+				throw new InvalidOperationException (string.Format ("Invalid framework: {0}", framework));
```
Comment (spouliot) [inline msbuild/Xamarin.MacDev.Tasks.Core/Sdks.cs:85]: not a new issue but... should not this be translated like other messages ?

**Review by spouliot — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by emaf — APPROVED** [review]

**Comment by mandel-macaque** [comment]
Test failures are unrelated: https://github.com/xamarin/maccore/issues/2357

File: msbuild/Xamarin.MacDev.Tasks.Core/Sdks.cs (hunk 1 above)
Comment (mandel-macaque) [reply msbuild/Xamarin.MacDev.Tasks.Core/Sdks.cs:85]: probably

**Review by mandel-macaque — APPROVED** [review]

**Review by tj-devel709 — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failures are known issues:

* Xamarin.MTouch.HttpClientHandler: https://github.com/xamarin/maccore/issues/2154
* Xamarin.MTouch.StripBitcodeFromFrameworks: fixed in #10400.

**Pull request merged**